### DISCUSSION TITLE
User Eloquent as customer

### DISCUSSION THREAD

**Discussion by mwagena** [duncanmcclean/simple-commerce#1093] [Help]
### Is it possible to use a Eloquent User as a customer?

To not have 2 different tables with auth logic;
I would like to have a Eloquent User be able to do orders, sign in and view it's orders.

Wat are the possibilities to achieve this?

Thanks in advance.

**Comment by duncanmcclean** [comment]
Sure, just store your customers as [Statamic users](https://simple-commerce.duncanmcclean.com/customers#content-users-driver) and [store the Statamic users in the database](https://statamic.dev/tips/storing-users-in-a-database).

**Reply by mwagena** [reply]
Hi Duncan, thanks, and when I want to store my orders in the database as well? How do we work with the relational aspect then?

**Reply by duncanmcclean** [reply]
When you switch orders to the database, it'll create a `customers` table for you and switch the "customer" driver to it. You're safe to just switch that "customer" driver back to the Statamic users driver and you can ignore the `customers` table.

**Reply by mwagena** [reply]
Getting there.. now i'm receiving the following error:
```
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'published' in 'field list'
```
It seems that it's trying to save a published field into my users table.

**Reply by mwagena** [reply]
Also it seems to have issues with creating orders on the User (model / tabel).

**Reply by duncanmcclean** [reply]
> It seems that it's trying to save a published field into my users table.

When you're storing users in a database, you'll need to create migrations for any fields you add to your blueprint. It's mentioned in [step 6 of the guide on the Statamic docs](https://statamic.dev/tips/storing-users-in-a-database).

> Also it seems to have issues with creating orders on the User (model / tabel).

What issues?

**Reply by mwagena** [reply]
About the 'published' column:
- it's caused by this piece of code in: https://github.com/duncanmcclean/simple-commerce/blob/7.x/src/Http/Controllers/Concerns/HandlesCustomerInformation.php#L74

The other issue is occurring when the mollie webhook tries to update the order, the following error is given:
```Column not found: 1054 Unknown column 'orders' in 'field list'```

**Reply by duncanmcclean** [reply]
Oh, I see. You might need to add the `published` column (a boolean) and an `orders` column (`json`).

**Reply by mwagena** [reply]
Hi Duncan, yeah that could be a solution but then we would work around the preferred way of setting relationships (by model relations). You had this tackled already with the Customer model and table. In that case i might take look in implementing it perhaps by a PR.
